packages/design-system/src/components/SearchBar/SearchBar.js

```diff
@@ -0,0 +1,199 @@
+import * as React from 'react';
+import * as PropTypes from 'prop-types';
+import cx from 'classnames';
+import SearchIcon from 'react-material-icon-svg/dist/MagnifyIcon';
+import CloseIcon from 'react-material-icon-svg/dist/CloseIcon';
+import { debounce } from '@livechat/data-utils';
+
+import styles from './style.scss';
+import { Loader } from '../Loader';
+import FieldError from '../FieldError';
+import { Input } from '../InputField';
+import { KeyCodes } from '../../constants/keyCodes';
+
+const baseClass = 'search-bar';
+const noop = () => {};
+
+export class SearchBar extends React.PureComponent {
+  constructor(props) {
+    super(props);
+
+    this.inputRef = React.createRef();
+    this.debouncedOnChange = debounce(this.props.debounceInMs, value => {
+      this.props.onChange(value);
+    });
+
+    this.state = {
+      searchTerm: '',
+      isInCompactMode: props.collapsable
+    };
+  }
+
+  handleChange = e => {
+    const { value } = e.target;
+    const { debounceInMs, onChange } = this.props;
+
+    this.setState({
+      searchTerm: value
+    });
+
+    if (debounceInMs) {
+      this.debouncedOnChange(value);
+    } else {
+      onChange(value);
+    }
+  };
+
+  handleKeyPress = event => {
+    const { onSubmit, onKeyDown } = this.props;
+
+    if (onKeyDown) {
+      onKeyDown(event);
+    }
+
+    if (onSubmit) {
+      if (event.keyCode === KeyCodes.enter) {
+        onSubmit(this.state.searchTerm);
+      }
+    }
+  };
+
+  handleClear = () => {
+    const { onChange, onSubmit, onClear } = this.props;
+    const searchFunction = onClear || onSubmit || onChange;
+
+    this.setState({
+      searchTerm: ''
+    });
+
+    // TODO: Cancel debounced search
+    // Add debounce and throttle from lodash as dependecy and use it in our Design System (after talking with @kamilmateusiak )
+    searchFunction('');
+
+    if (this.inputRef.current) {
+      this.inputRef.current.focus();
+    }
+  };
+
+  handleClearButtonKeyDown = event => {
+    if (event.keyCode === KeyCodes.enter) {
+      this.handleClear();
+    }
+  };
+
+  toggleCompactMode = () => {
+    if (this.props.collapsable) {
+      this.setState(
+        prevState => ({
+          isInCompactMode: !prevState.isInCompactMode
+        }),
+        () => {
+          if (!this.state.isInCompactMode && this.inputRef.current) {
+            this.inputRef.current.focus();
+          }
+        }
+      );
+    }
+  };
+
+  render() {
+    const {
+      className,
+      placeholder,
+      value,
+      loading,
+      collapsable,
+      error,
```

Comment:
in `restProps` you will have all remaining props. if you don't want to pass `debounceInMs` to the main `div` you should take it from it as you do for `collapsable` etc.